blog/2024-11-22-inspektor-gadget-unittesting-framework.md

```diff
@@ -0,0 +1,266 @@
+---
+authors: [sanskar-sharma]
+description: "Introducing the New Unit Testing Framework for Image-based Gadgets"
+draft: false
+tags:
+  [
+    "LFXMentorship",
+    "InspektorGadget",
+    "TestingFramework",
+    "eBPF",
+    "go",
+    "golang",
+    "gadget",
+    "tests",
+    "unittest",
+    "unit",
+  ]
+title: "Introducing a Unit Testing Framework for Gadgets"
+slug: /2024/11/inspektor-gadget-unittesting-framework
+image: /media/2024-05-31-cncf-mentoring-logo.jpg
+---
+
+Namaste Inspektors! I was selected by [Alban Crequy](https://github.com/alban) and [Mauricio Vásquez Bernal](https://github.com/mauriciovasquezbernal) under the LFX mentorship program to develop a Testing Framework for Image-based Gadgets to test them on different kernel versions(Issue [#3195](https://github.com/inspektor-gadget/inspektor-gadget/issues/3195) and [#1343](https://github.com/inspektor-gadget/inspektor-gadget/issues/1343)).
+
+In this blog, we’ll discuss the need for this framework and how to use it to effectively test your Gadgets. But first, let’s get familiar with Inspektor Gadget and Image-Based Gadgets.
+# What is Inspektor Gadget?
```

Comment:
Please add `<!-- truncate -->` to break this before this line. (Check other blog posts to see how it works)

Reply:
On It.